sale_project_milestone/models/sale_order_line.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2023 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    milestone_id = fields.Many2one(
+        "project.milestone",
+        string="Milestone",
+        copy=False,
+        index=True,
```

Comment:
@lanto-razafindrabe add attribute check_company=True to ensure that the milestone company is the one used in sol